app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+from __future__ import annotations
+from typing import Union
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.hidden = False
+        self.health = health
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{"
+                f"Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(other: Union[Carnivore, Herbivore]) -> None:
```

Comment:
at can be simlified to Animal, as all of them is child of Animal, don't needed Union